Purchases/Public/RCPurchases.m

```diff
@@ -624,36 +627,40 @@ - (void)restoreTransactionsWithCompletionBlock:(nullable RCReceivePurchaserInfoB
         [self.backend postReceiptData:data
                             appUserID:self.appUserID
                             isRestore:YES
-                    productIdentifier:nil
-                                price:nil
-                          paymentMode:RCPaymentModeNone
-                    introductoryPrice:nil
-                         currencyCode:nil
-                    subscriptionGroup:nil
-                            discounts:nil
+                          productInfo:nil
           presentedOfferingIdentifier:nil
                          observerMode:!self.finishTransactions
                  subscriberAttributes:subscriberAttributes
                            completion:^(RCPurchaserInfo *_Nullable info, NSError *_Nullable error) {
-                               [self dispatch:^{
-                                   if (error) {
-                                       [self markAttributesAsSyncedIfNeeded:subscriberAttributes
```

Comment:
some of the changes in this file are moving these completion blocks into functions for readability. the methods were pretty big otherwise, and it was hard to follow the flow